packages/facility-server/app/sync/FacilitySyncManager.js

```diff
@@ -156,11 +156,18 @@ export class FacilitySyncManager {
       startedAtTick: newSyncClockTime,
     });
 
-    await this.pushChanges(sessionId, newSyncClockTime);
-
-    await this.pullChanges(sessionId);
+    try {
+      await this.pushChanges(sessionId, newSyncClockTime);
 
-    await this.centralServer.endSyncSession(sessionId);
+      await this.pullChanges(sessionId);
+      await this.centralServer.endSyncSession(sessionId);
+    } catch (error) {
+      await this.centralServer.markSessionErrored(sessionId, error.message);
+      throw error;
+    } finally {
+      // clear temp data stored for persist
+      await dropSnapshotTable(this.sequelize, sessionId);
```

Comment:
So I assume this is desired? I think previously we wouldn't have cleaned this up if we hit an error during the persist phase, but that seems odd... unless we use it for debugging or something?

Reply:
No you're right to do this, can you check out central for the same issue? We've seen a big build up of snapshot tables there in some deployments (unless @passcod you've already followed up on that one)

Reply:
You were seeing it on central though hey? But yeah same pattern I imagine.